test/common/src/main/kotlin/org/codice/compliance/utils/decorators/IdpPostResponseDecorator.kt

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.utils.decorators
+
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.RELAY_STATE
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.SAML_RESPONSE
+import org.codice.compliance.saml.plugin.IdpPostResponse
+import org.codice.compliance.utils.TestCommon.Companion.acsUrl
+import org.codice.security.saml.SamlProtocol
+import com.jayway.restassured.path.xml.element.Node as raNode
+import org.w3c.dom.Node as w3Node
+
+class IdpPostResponseDecorator : IdpPostResponse {
+    companion object {
+        private const val HIDDEN = "hidden"
+        private const val TYPE = "type"
+        private const val ACTION = "action"
+        private const val METHOD = "method"
+        private const val POST = "POST"
+    }
+
+    // can only instantiate by using extension methods in IdpResponseDecorators.kt
+    internal constructor(response: IdpPostResponse): super(response)
+
+    var isRelayStateGiven: Boolean = false
+    lateinit var decodedSamlResponse: String
+    val responseDom: w3Node by lazy {
+        checkNotNull(decodedSamlResponse)
+        buildDom(decodedSamlResponse)
+    }
+
+    val isResponseFormNull: Boolean by lazy {
+        responseForm == null
+    }
+    val isSamlResponseFormNull: Boolean by lazy {
+        samlResponseForm == null
+    }
+    val isRelayStateFormNull: Boolean by lazy {
+        relayStateForm == null
+    }
+
+    val isSamlResponseNameCorrect: Boolean by lazy {
+        checkNodeAttribute(samlResponseForm, NAME, SAML_RESPONSE)
+    }
+
+    /*
+     * Bindings 3.5.4 "The action attribute of the form MUST be the recipient's HTTP endpoint for
+     * the protocol or profile using this binding to which the SAML message is to be delivered.
+     * The method attribute MUST be "POST"."
+     */
+    val isFormActionCorrect: Boolean by lazy {
+        checkNotNull(acsUrl[SamlProtocol.Binding.HTTP_POST])
+        checkNodeAttribute(responseForm, ACTION, acsUrl[SamlProtocol.Binding.HTTP_POST]!!)
```

Comment:
✏️ Instead of doing the `!!` right after checking for null, you could just do the `checkNotNull` method call on this same line. That method returns the non-null value.